Firestore/Swift/Source/Codable/third_party/FirestoreDecoder.swift

```diff
@@ -323,6 +330,17 @@ private struct _FirestoreKeyedDecodingContainer<K: CodingKey>: KeyedDecodingCont
   }
 
   public func decode<T: Decodable>(_: T.Type, forKey key: Key) throws -> T {
+    if T.self == AutoPopulatedDocumentId.self {
+      if contains(key) {
+        throw FirestoreDecodingError.fieldNameConfict("Field name conflict:" +
+          " \(key.stringValue) is an `AutoPopulatedDocumentId` but " +
+          "also exists as a field in the Firestore document being decoded.")
+      }
+      let docRef = decoder.userInfo[Firestore.Decoder.documentRefUserInfoKey!]
```

Comment:
Once you change the document reference to optional this may not be set so you you can't force unwrap the optional.